### DISCUSSION TITLE
Autocomplete mode for Listbox/Select component

### DISCUSSION THREAD

**Discussion by sergeyzwezdin** [tailwindlabs/headlessui#822] [Ideas]
It would be very helpful if Listbox would allow to build autocomplete like input when a user can type something and values in dropdown will be filtered out. Something like this:

![2021-09-18-09-43-14](https://user-images.githubusercontent.com/800755/133872773-3dc55a6c-1fd3-4389-8ce7-d244717e5808.png)

To do that, we need to update the `Listbox` component slightly:

1. Introduce `Listbox.Input` along with `Listbox.Button`, so it will allow to add `<input type="text" />`. The only difference here is that dropdown won't be focused when clicking out `Listbox.Input`.
2. Introduce search text (value of input) as a parameter for `OptionRenderPropArg` so we can use it when render option (to be able to highlight search value).

**Comment by sergeyzwezdin** [comment]
I can try to implement it for React component, but not for Vue.

**Comment by ranfdev** [comment]
By the way, the name of this UI pattern is called [combobox](https://en.wikipedia.org/wiki/Combo_box) and is present in most of the native GUI toolkits (for example, [Gtk](https://docs.gtk.org/gtk4/class.ComboBox.html)).

**Comment by Chaphasilor** [comment]
Related discussions: #806, #626
